web/src/views/application/components/ApplicationList.vue

```diff
@@ -0,0 +1,67 @@
+<!--
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+-->
+
+<script setup lang="ts">
+import type { PropType } from 'vue'
+import type { DataTableColumns } from 'naive-ui'
+import type { Application } from '@/api'
+import { normalizeDatetime } from '@/utils/datetime'
+
+defineOptions({
+  name: 'ApplicationDataTable'
+})
+
+defineProps({
+  data: {
+    type: Array as PropType<Application[]>,
+    default: () => []
+  }
+})
+
+const columns: DataTableColumns<Application> = [
+  {
+    title: 'ApplicationId',
+    key: 'appId'
+  },
+  {
+    title: 'SubUser',
+    key: 'subUser'
+  },
+  {
+    title: 'Tenant',
+    key: 'tenant'
+  },
+  {
+    title: 'HeartbeatTime',
+    key: 'heartbeatTime',
+    render: ({ heartbeatTime }) => normalizeDatetime(heartbeatTime)
+  },
+  {
+    title: 'ShuffleSize',
+    key: 'shuffleSize'
+  },
+  {
+    title: 'ActiveShuffle',
+    key: 'shuffleFileCount'
+  }
+]
+</script>
+
+<template>
+  <n-data-table :columns="columns" :data="data" :pagination="false" />
```

Comment:
Could `pagination` be true? BTW, the application list should support search and order, WDYT?

Reply:
> Could `pagination` be true? BTW, the application list should support search and order, WDYT?

@SteNicholas Now your interface doesn't support pagination, so the pagination property here is `false`.When the interface supports pagination in the future, we will modify it uniformly.

Regarding search and order, I don't think this feature will be very useful for the time being without pagination.

Reply:
@miyuesc, I don't agree above point. Search and order are the features of table, not pagination.

Reply:
Well, I'll add search and order later.
But with regard to pagination, we need to talk a little bit about the response data structure.

Reply:
@miyuesc, you could support search and order in this pull request and create another pull request to support pagination.

Reply:
okey